packages/query/src/stacks/fetch-wrapper.ts

```diff
@@ -0,0 +1,48 @@
+import axios from 'axios';
+
+// import { analytics } from '@shared/utils/analytics';
+
+const leatherHeaders: HeadersInit = {
+  // TODO #40: Need VERSION?
+  // 'x-leather-version': VERSION,
+  'x-hiro-product': 'leather',
+};
+
+function isErrorCode(statusCode: number) {
+  return statusCode >= 400;
+}
+
+function trackApiError(url: string, statusCode: number) {
+  // TODO #40: Use analytics here?
+  // eslint-disable-next-line no-console
+  console.log('api_error', { origin: new URL(url).origin, statusCode, url });
+  // void analytics.track('api_error', { origin: new URL(url).origin, statusCode, url });
+}
+
+/**
+ * @deprecated Use `axios` directly instead. Fetch only needed for interaction
+ * with generated stacks blockchain api library
+ */
+export async function wrappedFetch(input: RequestInfo, init: RequestInit = {}) {
+  const initHeaders = init.headers || {};
+  // eslint-disable-next-line no-restricted-globals
+  const resp = await fetch(input, {
+    ...init,
+    credentials: 'omit',
+    headers: { ...initHeaders, ...leatherHeaders },
+  });
+  if (isErrorCode(resp.status)) trackApiError(resp.url, resp.status);
+  return resp;
+}
```

Comment:
What do you suggest we do with this file? Would be nice not to commit it with loads of commented out code. 

Can you first describe the problem? Then solutions we might go for?

Reply:
@edgarkhanzadian am I right that adding the queryCache `onError` analytics would replace the need for using the `trackApiError` here?

Reply:
@kyranjamie it seems like our use of `wrappedFetch` is outdated? If we are only using it to support the `FetchApi` for the generated Stacks api, should we just remove our dependency on that library and use axios directly for Stacks queries? We appear to be already intercepting the axios request to add the proper Hiro headers.

Reply:
Pretty sure we pass this into the hiro client library, that's why it's needed, in order to add the query parms